File: cdk_integration_tests/src/typescript/SQSQueueEncryption/fail__2__.ts
import { App, Stack } from 'aws-cdk-lib';
import { Construct } from 'constructs';
import * as sqs from 'aws-cdk-lib/aws-sqs';

class SqsQueueWithKmsKeyStack extends Stack {
  constructor(scope: Construct, id: string, props?: {}) {
    super(scope, id, props);

    new sqs.Queue(this, "MySqsQueue", {
      encryption: sqs.QueueEncryption.KMS,
      visibilityTimeout: cdk.Duration.seconds(300) // Other properties for the queue
    });
  }
}

const app = new App();
new SqsQueueWithKmsKeyStack(app, "SqsQueueWithKmsKeyStack");
app.synth();

class SqsQueueWithKmsKeyIdStack extends Stack {
  constructor(scope: Construct, id: string, props?: {}) {
    super(scope, id, props);

    var mySqs = new sqs.CfnQueue(this, "MySqsQueue", {
      visibilityTimeout: 300  // Other properties for the queue
      // Specify the KMS key ID if needed here, e.g., kmsMasterKeyId: 'alias/aws/sqs'
    });
  }
}

const app2 = new App();
new SqsQueueWithKmsKeyIdStack(app2, "SqsQueueWithKmsKeyIdStack");
app2.synth();

File: cdk_integration_tests/src/python/SQSQueueEncryption/fail__2__.py
from aws_cdk import core
from aws_cdk import aws_sqs as sqs
class SqsQueueWithKmsKeyStack(core.Stack):

    def __init__(self, scope: core.Construct, id: str, **kwargs) -> None:
        super().__init__(scope, id, **kwargs)

        # Create an SQS queue with KMS encryption
        queue = sqs.Queue(self, "MySqsQueue",
            encryption=sqs.QueueEncryption.KMS,
            visibility_timeout=300  # Other properties for the queue
        )

app = core.App()
SqsQueueWithKmsKeyStack(app, "SqsQueueWithKmsKeyStack")
app.synth()



class SqsQueueWithKmsKeyIdStack(core.Stack):

    def __init__(self, scope: core.Construct, id: str, **kwargs) -> None:
        super().__init__(scope, id, **kwargs)

        # Define an SQS queue with a specific KmsMasterKeyId
        queue = sqs.CfnQueue(self, "MySqsQueue",
            visibility_timeout=300  # Other properties for the queue
        )

app = core.App()
SqsQueueWithKmsKeyIdStack(app, "SqsQueueWithKmsKeyIdStack")
app.synth()
